---
title: "編譯器錯誤 CS0156"
ms.date: 07/20/2015
ms.prod: .net
ms.technology:
- devlang-csharp
ms.topic: article
f1_keywords:
- CS0156
helpviewer_keywords:
- CS0156
ms.assetid: 32026b1b-bcd7-4464-b63f-3b38c00452a6
caps.latest.revision: 
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: f755172711f7a09bcbdae518fdcb82aca58cad20
ms.sourcegitcommit: 83dd5ec003e788ccb3e33f3412a7af39ae347646
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 03/15/2018
---
# <a name="compiler-error-cs0156"></a>編譯器錯誤 CS0156
最內層 catch 子句中巢狀 finally 子句不允許沒有引數的 throw 陳述式  
  
 沒有任何參數的 [throw](../../csharp/language-reference/keywords/throw.md) 陳述式只會出現在不接受任何參數的 **catch** 子句中。  
  
 如需詳細資訊，請參閱[例外狀況處理陳述式](../../csharp/language-reference/keywords/exception-handling-statements.md)和[例外狀況和例外狀況處理](../../csharp/programming-guide/exceptions/index.md)。  
  
 下列範例會產生 CS0156：  
  
```csharp  
// CS0156.cs  
using System;  
  
namespace MyNamespace  
{  
   public class MyClass2 : Exception  
   {  
   }  
  
   public class MyClass  
   {  
      public static void Main()  
      {  
         try  
         {  
            throw;   // CS0156  
         }  
  
         catch(MyClass2)  
         {  
            throw;   // this throw is valid  
         }  
      }  
   }  
}  
```
